File: TDS/Models/Clase.cs
using System;
using System.Collections.Generic;

namespace TDS.Models
{
    public partial class Clase
    {
        public Clase()
        {
            EstudiantesClases = new HashSet<EstudiantesClase>();
            Mensajes = new HashSet<Mensaje>();
            MensajesDetalles = new HashSet<MensajesDetalle>();
            Tareas = new HashSet<Tarea>();
        }

        public int Id { get; set; }
        public string? Nombre { get; set; }
        public string? Descripcion { get; set; }
        public int? MaestroId { get; set; }
        public int? InstitucionId { get; set; }
        public string? Codigo { get; set; }
        public bool? Estado { get; set; }

        public virtual Institucion? Institucion { get; set; }
        public virtual Maestro? Maestro { get; set; }
        public virtual ICollection<EstudiantesClase> EstudiantesClases { get; set; }
        public virtual ICollection<Mensaje> Mensajes { get; set; }
        public virtual ICollection<MensajesDetalle> MensajesDetalles { get; set; }
        public virtual ICollection<Tarea> Tareas { get; set; }
    }
}

File: TDS_Front/src/app/models/clase.ts
import { EstudiantesClases } from "./estudiantes-clases";
import { Maestro } from "./maestro";
import { Tarea } from "./tarea";

export class Clase {
  constructor(public id: number = 0, public nombre: string = "", public descripcion: string = "", public maestroId: number = 0, public institucionId: number = 0, public codigo: string = "", public estado: boolean = false, public horaInicio: string = "", public horaFin: string = "", public maestro: Maestro= new Maestro(), public estudiantesClases: EstudiantesClases[] = [], public tareas: Tarea[] = []){}
}
